### PR TITLE
Fix lint in deployment package

### PR DESCRIPTION
**Pull request by michaelfig**
A mostly-mechanical transformation, and enable the CI lint-check for deployment.

### CODE REVIEW

File: packages/deployment/main.js [hunk 1]
```diff
@@ -962,39 +981,38 @@ ${units}`;
...
-      if (cmd.length === 1) {
-        if (cmd[0].match(shellMetaRegexp)) {
+      if (subcmd.length === 1) {
+        if (subcmd[0].match(shellMetaRegexp)) {
           // Already contains metacharacters.
-          runArg = `sh -c ${shellEscape(cmd[0])}`;
+          runArg = `sh -c ${shellEscape(subcmd[0])}`;
         } else {
-          runArg = cmd[0];
+          [runArg] = subcmd;
```
Comment (FUDCo) [inline packages/deployment/main.js:1015]: This kind of thing inclines me towards thinking our default lint rc should remove the `prefer-destructuring` rule.

**Review by FUDCo — APPROVED** [review]
I feel your pain.

File: packages/deployment/main.js (hunk 1 above)
Comment (erights) [reply packages/deployment/main.js:1015]: The es-lint rules should agree across all our code unless there's a very good reason for them to differ. We should consider changes to our rules to apply across our entire code base, to be applied automatically and, ideally, atomically. Killing `prefer-destructuring` should be on the table then, as should `"trailing-comma": "all"`.

But we should not consider making any such changes only for one package.

File: packages/deployment/init.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable no-continue */
+/* eslint-disable no-await-in-loop */
```
Comment (Chris-Hibbert) [inline packages/deployment/init.js:2]: Placing these exceptions here makes me uncomfortable. We should only enforce these rules if we feel strongly enough that they're the right convention that we want to them to apply in all but exceptional circumstances. We should either require documenting why an exception is necessary in a particular case, or drop the rule.

I think no-continue is a fine rule, and I would favor enforcing it here. 

My impression is that no-await-in-loop is generally a good rule, and exceptions should be documented locally where they occur and not globally over a file.

Comment (erights) [reply packages/deployment/init.js:2]: I can't remember the last time I used a `continue` so I haven't thought much about it. What's wrong with using `continue`?

Comment (Chris-Hibbert) [reply packages/deployment/init.js:2]: I don't have a specific problem with `continue`. I use it occasionally. The uses in this file were trivially replaced producing clearer code. If someone thinks that's always the case, I'm willing to try and willing to override with an explicit explanation when the rule is wrong.

**Pull request merged**